client/src/components/Donation/DonateForm.tsx

```diff
@@ -72,8 +102,12 @@ const mapDispatchToProps = {
   updateDonationFormState
 };
 
-class DonateForm extends Component {
-  constructor(...args) {
+class DonateForm extends Component<DonateFormProps, DonateFormState> {
+  static displayName = 'DonateForm';
+  durations: { month: string; onetime: string };
```

Comment:
```suggestion
  durations: { month: 'monthly'; onetime: 'one-time' };
```
The assignment will need this type annotation too. I'll add it below.